src/libs/ajax/AzureStorage.ts

```diff
@@ -111,6 +111,51 @@ export const AzureStorage = (signal?: AbortSignal) => ({
     }), notebooks)
   },
 
+  blobMetadata: (azureStorageUrl: string) => {
+    const getObjectMetadata = async () => {
+      const workspaceId = azureStorageUrl.split('/')[3].replace('sc-', '')
+      const fileName = _.last(azureStorageUrl?.split('/'))?.split('.')?.join('.')
+      // check if file can just be fetched without sas token
+      try {
+        const res = await fetchOk(azureStorageUrl)
+        const text = await res.headers.entries()
+        const dict = {}
+        for (const pair of text) {
+          dict[pair[0]] = pair[1]
+        }
+
+        return { lastModified: dict['last-modified'], size: dict['content-length'], azureStorageUrl, workspaceId, fileName }
+      } catch (e) {
+        // file not public, proceed to try and get it with sas token
+      }
+
+      // assumption is made that container name guid in uri always matches the workspace Id guid it is present in
+      const { sas: { token } } = await AzureStorage(signal).details(workspaceId)
+
+      // instead of taking the url returned by azure storage, take it from the incoming url since there may be a folder path
+      const urlwithFolder = azureStorageUrl.substring(0, azureStorageUrl.lastIndexOf('/'))
+      const azureSasStorageUrl = _.flow(
+        _.split('?'),
+        _.head,
+        Utils.append(`/${fileName}?&${token}`),
+        _.join('')
+      )(urlwithFolder)
+
+      const res = await fetchOk(azureSasStorageUrl)
+      const text = await res.headers.entries()
+      const dict = {}
+      for (const pair of text) {
+        dict[pair[0]] = pair[1]
+      }
+
+      return { lastModified: dict['last-modified'], size: dict['content-length'], azureSasStorageUrl, workspaceId, fileName }
+    }
+
+    return {
+      getData: getObjectMetadata
+    }
+  },
+
   blob: (workspaceId: string, blobName: string) => {
```

Comment:
Would it make sense to make `getMetadata` a method on the object returned from `blob`?

Reply:
blob by default takes in a workspace Id and blob name -- I was thinking it may be nice to have an option to get data just using the url, that way the parsing out of workspace Id/folder/file name happens in one place vs having to do that in all places that would want to get blob metadata with a blob url (which seems to be the most common case)

Reply:
Fair enough. I think ideally we'd have multiple ways to construct the same Blob object, either with a workspace ID and blob name or a URL. Instead of having different methods available for the blob depending on which way it was referenced.

Reply:
after thinking about this more, I am torn...I definitely get the advantage of having a single blob object, butit becomes confusing where you have an object that can have parameters that are optional and it functions differently based on the params provided.

Reply:
as per our meeting conversation, it would be ideal to have a single blob object but this is a larger refactor that should be part of a different PR